packages/ui/src/primitives/tailwind/Color/index.tsx

```diff
@@ -52,36 +52,68 @@ export function ColorInput({
 }: ColorInputProp) {
   const color = new Color(value)
   const hexColor = '#' + color.getHexString()
+  const [isPickerOpen, setIsPickerOpen] = useState(false)
+  const pickerRef = useRef<HTMLDivElement>(null)
+  const inputRef = useRef<HTMLDivElement>(null)
+
+  const handleTogglePicker = () => {
+    setIsPickerOpen(!isPickerOpen) // Toggle visibility
+    onRelease && onRelease(color)
+  }
 
   const handleChange = (result: ColorResult) => {
     const color = new Color(result.hex)
     onChange(color)
   }
+
+  // Close the picker if clicking outside the input or picker - this method didn't require a new signature
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (
+        inputRef.current &&
+        !inputRef.current.contains(event.target) &&
+        pickerRef.current &&
+        !pickerRef.current.contains(event.target)
+      ) {
+        setIsPickerOpen(false)
+        if (onRelease) onRelease(color)
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside)
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside)
+    }
+  }, [hexColor, onRelease])
+
   return (
     <div
+      tabIndex={0} // Make the div focusable
+      ref={inputRef} // Ref to track focus on the input
       className={twMerge(
         'relative flex h-9 items-center gap-1 rounded-lg border-none bg-[#1A1A1A] px-2 text-xs text-[#8B8B8D]',
         disabled && 'cursor-not-allowed',
         className
       )}
+      onClick={handleTogglePicker} //opens the color picker
```

Comment:
I don't think this should be a toggle, it will also be called when the color picker itself is clicked, so the color picker will be closed every time you drag the color selector and release

Reply:
changed